File: clients/csharp/Model/LootBox.cs
namespace AiCup2019.Model
{
    public struct LootBox
    {
        public Model.Vec2Double Position { get; set; }
        public Model.Vec2Double Size { get; set; }
        public Model.Item Item { get; set; }
        public LootBox(Model.Vec2Double position, Model.Vec2Double size, Model.Item item)
        {
            this.Position = position;
            this.Size = size;
            this.Item = item;
        }
        public static LootBox ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new LootBox();
            result.Position = Model.Vec2Double.ReadFrom(reader);
            result.Size = Model.Vec2Double.ReadFrom(reader);
            result.Item = Model.Item.ReadFrom(reader);
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            Position.WriteTo(writer);
            Size.WriteTo(writer);
            Item.WriteTo(writer);
        }
    }
}

File: clients/fsharp/Model/LootBox.fs
#nowarn "0058"
namespace AiCup2019.Model
type LootBox = {
    Position: Vec2Double;
    Size: Vec2Double;
    Item: Item;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        this.Position.writeTo writer
        this.Size.writeTo writer
        this.Item.writeTo writer
    static member readFrom(reader: System.IO.BinaryReader) = {
        Position = Vec2Double.readFrom reader
        Size = Vec2Double.readFrom reader
        Item = Item.readFrom reader
    }
